test/Algebra/Graph/Test/RewriteRules.hs

```diff
@@ -18,81 +18,76 @@ import Algebra.Graph.Internal
-transposeEmpty = transpose Empty
+inspect $ 'clique1 === 'cliqueR
 
-inspect $ 'empty' === 'transposeEmpty
+-- Testsuite for 'transpose'.
+empty1, emptyR :: Graph a
+empty1 = transpose Empty
+emptyR = Empty
 
---- transpose . vertex
-vertex', transposeDotVertex :: a -> Graph a
-vertex'            = Vertex
-transposeDotVertex = transpose . vertex
+inspect $ 'empty1 === 'emptyR
 
-inspect $ 'vertex' === 'transposeDotVertex
+vertex1, vertexR :: a -> Graph a
+vertex1 = transpose . vertex
+vertexR = Vertex
 
---- transpose . overlay
-overlayTransposed, transposeDotOverlay :: Graph a -> Graph a -> Graph a
-overlayTransposed   g1 g2 = Overlay (transpose g1) (transpose g2)
-transposeDotOverlay g1 g2 = transpose $ Overlay g1 g2
+inspect $ 'vertex1 === 'vertexR
 
-inspect $ 'overlayTransposed === 'transposeDotOverlay
+overlay1, overlayR :: Graph a -> Graph a -> Graph a
+overlay1 x y = transpose (Overlay x y)
+overlayR x y = Overlay (transpose x) (transpose y)
 
---- transpose . connect
-connectTransposed, transposeDotConnect :: Graph a -> Graph a -> Graph a
-connectTransposed   g1 g2 = Connect (transpose g2) (transpose g1)
-transposeDotConnect g1 g2 = transpose $ Connect g1 g2
+inspect $ 'overlay1 === 'overlayR
 
-inspect $ 'connectTransposed === 'transposeDotConnect
+connect1, connectR :: Graph a -> Graph a -> Graph a
+connect1 x y = transpose (Connect x y)
+connectR x y = Connect (transpose y) (transpose x)
 
---- transpose . overlays
-overlaysTransposed, transposeDotOverlays :: [Graph a] -> Graph a
-overlaysTransposed   = overlays . map transpose
-transposeDotOverlays = transpose . overlays
+inspect $ 'connect1 === 'connectR
 
-inspect $ 'overlaysTransposed === 'transposeDotOverlays
+overlays1, overlaysR :: [Graph a] -> Graph a
+overlays1 = transpose . overlays
+overlaysR = overlays . map transpose
 
---- transpose . connects
-connectsTransposed, transposeDotConnects :: [Graph a] -> Graph a
-connectsTransposed   = fromMaybe Empty . foldr (maybeF (flip Connect) . transpose) Nothing
-transposeDotConnects = transpose . connects
+inspect $ 'overlays1 === 'overlaysR
 
-inspect $ 'connectsTransposed === 'transposeDotConnects
+connects1, connectsR :: [Graph a] -> Graph a
+connects1 = transpose . connects
+connectsR = fromMaybe Empty . foldr (maybeF (flip Connect) . transpose) Nothing
 
---- transpose . vertices
-transposeDotVertices :: [a] -> Graph a
-transposeDotVertices = transpose . overlays . map vertex
+inspect $ 'connects1 === 'connectsR
 
-inspect $ 'vertices' === 'transposeDotVertices
+vertices2 :: [a] -> Graph a
+vertices2 = transpose . overlays . map vertex
 
---- transpose . clique
-cliqueTransposed, transposeDotClique :: [a] -> Graph a
-cliqueTransposed   = fromMaybe Empty . foldr (maybeF (flip Connect) . Vertex) Nothing
-transposeDotClique = transpose . connects . map vertex
+inspect $ 'vertices2 === 'vertices1
 
-inspect $ 'cliqueTransposed === 'transposeDotClique
+-- TODO: Why can't we reach the desired target expression?
```

Comment:
About your todo,

The CSE is replacing the body of `verticesR` with only `vertices1`:
```
        verticesR :: forall a. [a] -> Graph a
        [LclIdX,                   
         Arity=1,                  
         Str=<S,1*U>,              
         Unf=Unf{Src=<vanilla>, TopLvl=True, Value=True, ConLike=True,
                 WorkFree=True, Expandable=True,
                 Guidance=ALWAYS_IF(arity=0,unsat_ok=True,boring_ok=True)}]
        verticesR = vertices1 
```

and GHC will never inline this definition so they are declared as different (I have reported this "bug" for inspection-testing).

Reply:
Ah, yes, thanks for clarifying. I thought that was related to the CSE issue, but couldn't quite connect all dots.

Reply:
So, we have a strange non-transitive situation:

```haskell
inspect $ 'vertices1 === 'verticesR
inspect $ 'vertices2 === 'vertices1
inspect $ 'vertices2 =/= 'verticesR
```
I guess I'll add this non-transitivity example to your `inspection-testing` issue.

Reply:
I've updated the comment to:
```
-- Note that we currently have these three tests:
-- * vertices2 === vertices1
-- * vertices1 === verticesR
-- * vertices2 =/= verticesR
-- This non-transitivity is awkward, and feels like a bug in the inspection
-- testing library. See https://github.com/nomeata/inspection-testing/issues/23.
```